core/services/chainlink/relayer_chain_interoperators.go

```diff
@@ -0,0 +1,367 @@
+	}
+	sort.Slice(relayerIds, func(i, j int) bool {
+		return relayerIds[i].String() < relayerIds[j].String()
+	})
+	for _, rid := range relayerIds {
+		lr := rs.loopRelayers[rid]
+		// the relayer is chain specific; use the chain id and not the relayer id
+		stat, err := lr.ChainStatus(ctx, rid.ChainID.String())
+		if err != nil {
+			totalErr = errors.Join(totalErr, err)
+			continue
+		}
+		stats = append(stats, stat)
+	}
+
+	if totalErr != nil {
+		return nil, 0, totalErr
+	}
+	cnt := len(stats)
+	if len(stats) > limit+offset && limit > 0 {
+		return stats[offset : offset+limit], cnt, nil
+	}
+	return stats[offset:], cnt, nil
+}
+
+func (rs *CoreRelayerChainInteroperators) Node(ctx context.Context, name string) (types.NodeStatus, error) {
+	// This implementation is round-about
+	// TODO BFC-2511, may be better in the loop.Relayer interface itself
+	stats, _, err := rs.NodeStatuses(ctx, 0, -1)
+	if err != nil {
+		return types.NodeStatus{}, err
+	}
+	for _, stat := range stats {
+		if stat.Name == name {
+			return stat, nil
+		}
+	}
+	return types.NodeStatus{}, chains.ErrNotFound
+}
+
+// ids must be a string representation of relay.Identifier
+// ids are a filter; if none are specificied, all are returned.
+// TODO: BCF-2440/1 this signature can be changed to id relay.Identifier which is a much better API
+func (rs *CoreRelayerChainInteroperators) NodeStatuses(ctx context.Context, offset, limit int, relayerIDs ...string) (nodes []types.NodeStatus, count int, err error) {
+	var (
+		totalErr error
+		result   []types.NodeStatus
+	)
+	if len(relayerIDs) == 0 {
+		for rid, lr := range rs.loopRelayers {
+			stats, _, err := lr.NodeStatuses(ctx, offset, limit, rid.ChainID.String())
+			if err != nil {
+				totalErr = errors.Join(totalErr, err)
+				continue
+			}
+			result = append(result, stats...)
+		}
+	} else {
+		for _, idStr := range relayerIDs {
+			rid := new(relay.Identifier)
+			err := rid.UnmarshalString(idStr)
+			if err != nil {
+				totalErr = errors.Join(totalErr, err)
+				continue
+			}
+			lr, exist := rs.loopRelayers[*rid]
+			if !exist {
+				totalErr = errors.Join(totalErr, fmt.Errorf("relayer %s does not exist", rid.Name()))
+				continue
+			}
+			nodeStats, _, err := lr.NodeStatuses(ctx, offset, limit, rid.ChainID.String())
+
+			if err != nil {
+				totalErr = errors.Join(totalErr, err)
+				continue
+			}
+			result = append(result, nodeStats...)
+		}
+	}
+	if totalErr != nil {
+		return nil, 0, totalErr
+	}
+	if len(result) > limit && limit > 0 {
+		return result[offset : offset+limit], limit, nil
+	}
+	return result[offset:], len(result[offset:]), nil
+}
+
+type FilterFn func(id relay.Identifier) bool
+
+var AllRelayers = func(id relay.Identifier) bool {
+	return true
+}
+
+// Returns true if the given network matches id.Network
+func FilterByType(network relay.Network) func(id relay.Identifier) bool {
+	return func(id relay.Identifier) bool {
+		return id.Network == network
+	}
+}
```

Comment:
Would it make sense to move these to `package relay`?
Otherwise, they could use some extra context since they are exported, e.g. `RelayIDFilterFn` :shrug:

Reply:
expanded the name. i tried moving it but that introduces a circular dependency.